src/ExportParameters.cpp

```diff
@@ -0,0 +1,171 @@
+                frame.sourceID == variable_speed::DDC_CONTROLLER_ADDRESS) {
+                return frame;
+            }
+        }
+        catch (const variable_speed::WrongSize& e) {
+        }
+        catch (const variable_speed::InvalidChecksum& e) {
+        }
+        // iodrivers_base may throw this error when receiving a SIGINT, but it can be
+        // ignored
+        catch (const iodrivers_base::UnixError& e) {
+        }
+    }
+    // Never reached
+}
+
+int main(int argc, char** argv)
+{
+    list<string> args(argv + 1, argv + argc);
+    if (args.size() < 2) {
+        bool error = !args.empty();
+        usage(error ? cerr : cout);
+        return error;
+    }
+    string uri = args.front();
+    args.pop_front();
+
+    if (args.empty()) {
+        usage(cerr);
+        return 1;
+    }
+
+    string cmd = args.front();
+    args.pop_front();
+
+    unique_ptr<VariableSpeedMaster> master(new VariableSpeedMaster());
+    master->openURI(uri);
+
+    if (cmd == "run") {
+        if (args.size() != 0) {
+            cerr << "command run does not accept arguments\n\n";
+            usage(cerr);
+            return 1;
+        }
+
+        waitForValidFrame(master);
+
+        master->sendControlCommand(CONTROL_CMD_START);
+
+        signal(SIGINT, handler);
+
+        // Send keep_alive until user presses Ctrl+C
+        while (!stop) {
+            waitForValidFrame(master);
+
+            try {
+                master->sendControlCommand(CONTROL_CMD_KEEP_ALIVE);
+            }
+            catch (const iodrivers_base::UnixError& e) {
+            } // iodrivers_base may throw this error when receiving a SIGINT, but it can
+              // be ignored
+        }
+
+        waitForValidFrame(master);
+
+        Frame frame;
+        base::Time now;
+        pair<GeneratorState, GeneratorModel> generatorStateAndModel;
+        RunTimeState runTimeState;
+        bool receivedGeneratorState = false;
+        bool receivedRunTimeState = false;
+
+        while (!(receivedGeneratorState && receivedRunTimeState)) {
+            frame = waitForValidFrame(master);
+
+            now = base::Time::now();
+            if (frame.command == variable_speed::PACKET_GENERATOR_STATE_AND_MODEL) {
+                generatorStateAndModel =
+                    master->parseGeneratorStateAndModel(frame.payload, now);
+                receivedGeneratorState = true;
+            }
+            else if (frame.command == variable_speed::PACKET_RUN_TIME_STATE) {
+                runTimeState = master->parseRunTimeState(frame.payload, now);
+                receivedRunTimeState = true;
+            }
+        }
+
+        writeAndExportParameters(generatorStateAndModel, runTimeState);
+        cout << generatorStateAndModel.first << endl;
+        cout << generatorStateAndModel.second << endl;
+        cout << runTimeState << endl;
+
+        master->sendControlCommand(CONTROL_CMD_STOP);
```

Comment:
Uncouple the parameters exportation from the run cmd. The user should be able to use it like this:

```
./script_name run
./script_name info
./script_name stop
```